daemon/src/model/cfd.rs

```diff
@@ -240,53 +250,45 @@ impl Display for CfdState {
 /// Represents a cfd (including state)
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 pub struct Cfd {
-    pub order_id: OrderId,
-    pub initial_price: Usd,
-
-    pub leverage: Leverage,
-    pub trading_pair: TradingPair,
-    pub position: Position,
-    pub liquidation_price: Usd,
-
+    pub order: Order,
     pub quantity_usd: Usd,
-
     pub state: CfdState,
+    /* TODO: Leverage is currently derived from the Order, but the actual leverage should be
+     * stored in the Cfd once there is multiple choices of leverage */
 }
 
 impl Cfd {
-    pub fn new(cfd_order: Order, quantity: Usd, state: CfdState, position: Position) -> Self {
+    pub fn new(order: Order, quantity: Usd, state: CfdState) -> Self {
         Cfd {
-            order_id: cfd_order.id,
-            initial_price: cfd_order.price,
-            leverage: cfd_order.leverage,
-            trading_pair: cfd_order.trading_pair,
-            position,
-            liquidation_price: cfd_order.liquidation_price,
+            order,
             quantity_usd: quantity,
             state,
         }
     }
 
     pub fn calc_margin(&self) -> Result<Amount> {
-        let margin = match self.position {
+        let margin = match self.position() {
             Position::Buy => {
-                calculate_buy_margin(self.initial_price, self.quantity_usd, self.leverage)?
+                calculate_buy_margin(self.order.price, self.quantity_usd, self.order.leverage)?
             }
-            Position::Sell => calculate_sell_margin(self.initial_price, self.quantity_usd)?,
+            Position::Sell => calculate_sell_margin(self.order.price, self.quantity_usd)?,
         };
 
         Ok(margin)
     }
 
     pub fn calc_profit(&self, current_price: Usd) -> Result<(Amount, Usd)> {
-        let profit = calculate_profit(
-            self.initial_price,
-            current_price,
-            dec!(0.005),
-            Usd(dec!(0.1)),
-        )?;
+        let profit =
+            calculate_profit(self.order.price, current_price, dec!(0.005), Usd(dec!(0.1)))?;
         Ok(profit)
     }
+
+    pub fn position(&self) -> Position {
+        match self.order.origin {
+            Origin::Ours => self.order.position.clone(),
+            Origin::Theirs => self.order.position.counter_position(),
+        }
```

Comment:
with the `position` logic now nicely encapsulated, we probably don't need to leak `counter_position` in its public interface - or are you planning to use it anywhere?

Reply:
not for the moment, good catch, I'll remove it